src/frontend/src/catalog/schema_catalog.rs

```diff
@@ -52,6 +52,10 @@ pub struct SchemaCatalog {
     connection_by_name: HashMap<String, Arc<ConnectionCatalog>>,
     connection_by_id: HashMap<ConnectionId, Arc<ConnectionCatalog>>,
 
+    // This field is currently used only for `show connections`
+    connection_source_ref: HashMap<ConnectionId, Vec<SourceId>>,
```

Comment:
Considering source and sink may not be dropped frequently, so it is ok to use a `Vec` to store those source ids.